test/unit/test-pure-assert.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  USER_ERROR_SENTINEL,
+  pureDevAssert as devAssert,
+  pureUserAssert as userAssert,
+} from '../../src/pure-assert';
+
+describe('assertions', () => {
+  describe('devAssert', () => {
+    it('should not fail for truthy values', () => {
+      devAssert(true, 'True!');
+      devAssert(1, '1');
+      devAssert('abc', 'abc');
+    });
+
+    it('should fail for falsey values dev', () => {
+      expect(() => devAssert(false, 'xyz')).to.throw('xyz');
+      expect(() => userAssert(false, '123')).to.throw(
+        `123${USER_ERROR_SENTINEL}`
+      );
+    });
+  });
+
+  describe('userAssert', () => {
+    it('should not fail for truthy values', () => {
+      userAssert(true, 'True!');
+      userAssert(1, '1');
+      userAssert('abc', 'abc');
+    });
+
+    it('should fail for falsey values dev', () => {
+      expect(() => userAssert(false, 'xyz')).to.throw(
```

Comment:
Do these not need to be wrapped with `allowConsoleError`?

Reply:
As written, the assert helpers don't console log, they just throw the error. It wasn't clear to me that it was logged before, but inspecting further I see it is. `console.assert()` could easily be used here, but do we think this is still necessary if the assertion throws an errer?

Since this already throws an error, that error is either a) unhandled, and reaches console anyway or b) caught and handled, in which case it would make sense for the catching code to handle/optionally log or swallow the error. Thoughts?